Libraries/Inspector/Inspector.js

```diff
@@ -126,43 +150,47 @@ class Inspector extends React.Component {
     });
   };
 
+
   setSelection(i: number) {
-    const instance = this.state.hierarchy[i];
-    // if we inspect a stateless component we can't use the getPublicInstance method
-    // therefore we use the internal _instance property directly.
-    const publicInstance = instance['_instance'] || {};
-    const source = instance['_currentElement'] && instance['_currentElement']['_source'];
-    UIManager.measure(instance.getHostNode(), (x, y, width, height, left, top) => {
+    const hierarchyItem = this.state.hierarchy[i];
+    // we pass in ReactNative.findNodeHandle as the method is injected
+    const {
+      measure,
+      props,
+      source,
+    } = hierarchyItem.getInspectorData(ReactNative.findNodeHandle);
+
+    measure((x, y, width, height, left, top) => {
       this.setState({
         inspected: {
           frame: {left, top, width, height},
-          style: publicInstance.props ? publicInstance.props.style : {},
+          style: props ? props.style : {},
```

Comment:
It's a bit confusing it's undefined when `props` is `emptyObject` but an empty object if props is undefined. It's better if we either always make it an `emptyObject` when it's missing, or always make it `null` when it's missing.